library-udf/src/test/java/org/apache/iotdb/library/string/StringTests.java

```diff
@@ -0,0 +1,209 @@
+      statement.execute(
+          "create function regexreplace as 'org.apache.iotdb.library.string.UDTFRegexReplace'");
+      statement.execute(
+          "create function regexsplit as 'org.apache.iotdb.library.string.UDTFRegexSplit'");
+      statement.execute(
+          "create function strreplace as 'org.apache.iotdb.library.string.UDTFStrReplace'");
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    EnvFactory.getEnv().cleanAfterClass();
+    ConfigFactory.getConfig()
+        .setUdfCollectorMemoryBudgetInMB(oldUdfCollectorMemoryBudgetInMB)
+        .setUdfTransformerMemoryBudgetInMB(oldUdfTransformerMemoryBudgetInMB)
+        .setUdfReaderMemoryBudgetInMB(oldUdfReaderMemoryBudgetInMB);
+  }
+
+  @Test
+  public void testRegexMatch1() {
+    String sqlStr =
+        "select regexmatch(d1.s1,\"regex\"=\"\\d+\\.\\d+\\.\\d+\\.\\d+\", \"group\"=\"0\") from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      resultSet.last();
+      int resultSetLength = resultSet.getRow();
+      Assert.assertEquals(resultSetLength, 5);
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testRegexReplace1() {
+    String sqlStr =
+        "select regexreplace(d2.s1,\"regex\"=\"192\\.168\\.0\\.(\\d+)\", \"replace\"=\"cluster-$1\") from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      resultSet.last();
+      int resultSetLength = resultSet.getRow();
+      Assert.assertEquals(resultSetLength, 5);
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testRegexSplit1() {
+    String sqlStr = "select regexsplit(d2.s1, \"regex\"=\",\") from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      resultSet.last();
+      int resultSetLength = resultSet.getRow();
+      Assert.assertEquals(resultSetLength, 5);
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testStrReplace1() {
+    String sqlStr =
+        "select strreplace(d2.s1,\"target\"=\",\", \"replace\"=\"_\") from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      resultSet.last();
+      int resultSetLength = resultSet.getRow();
```

Comment:
The test cases should check whether the returned data is as our expected, not only row count.

Reply:
Done.